### ISSUE TITLE
Running with --aspects flag stops outputting output binary paths

### ISSUE THREAD

**Issue by illicitonion** [bazelbuild/bazel#18937]
### Description of the bug:

If you build some target, you get as output the path that was built:
```console
% bazel build :bin                                                         
INFO: Invocation ID: 4e80f1e3-18eb-4757-a045-dcc16a5df0e2
INFO: Analyzed target //:bin (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //:bin up-to-date:
  bazel-bin/bin
INFO: Elapsed time: 0.089s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
```

If you add an `--aspects` flag (e.g. to perform additional linting), but are still generating your default outputs, those paths no longer get output to the terminal:

```console
% bazel build :bin --aspects=@rules_rust//rust:defs.bzl%rust_clippy_aspect 
INFO: Invocation ID: ff889c33-97a9-44df-9773-14dff1f33f05
INFO: Analyzed target //:bin (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Aspect @rules_rust//rust/private:clippy.bzl%rust_clippy_aspect of //:bin up-to-date (nothing to build)
INFO: Elapsed time: 0.082s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
```

The file was still built to the same path, it's just not output, so it's hard to tell as a person doing a build where the output is that you may want to run or inspect.

This is causing a problem because in a repo I own, we added some `--aspects` flags to our default `.bazelrc` to perform linting, but this has made it harder to work in the repo because e.g. `bazel build`ing is less informative.

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Run a `bazel build` for a single binary with an `--aspects` flag set.

### Which operating system are you running Bazel on?

macOS

### What is the output of `bazel info release`?

release 6.2.1

**Comment by meisterT** [comment]
cc @mai93

**Comment by mai93** [comment]
I could not reproduce it with bazel@head I think it is fixed with 944b154135444813a6eed464b19b4cc6d71dde73

**Comment by stagnation** [comment]
Yepp, we saw this in a recent build
We can now see both the rule outputs and the aspects when using the default output group.
At first it was a little confusing because it now counts as two targets, for `--show_result=1`. So no output was shown with an aspect enabled, using `--show_result=2` shows it just fine.
